src/tcp.h

```diff
@@ -22,9 +22,10 @@
 struct enftun_tcp
 {
     int fd; // file descriptor for the underlying TCP socket
+    struct sockaddr local_addr;
```

Comment:
I think that `struct sockaddr` is too small to hold all of a `struct sockaddr_in` or `struct sockaddr_in6`.   I suggest declaring it like this:
```
#define MAX_SOCKADDR_LEN sizeof(struct sockaddr_in6)

struct enftun_tcp
{
    int fd;

    union {
        struct sockaddr local_addr;
        char _local_addr_pad[MAX_SOCKADDR_LEN];
    };

   union {
        struct sockaddr remote_addr;
        char _remote_addr_pad[MAX_SOCKADDR_LEN];
   };
}
```

Now, enough space is reserved for the largest (`sockaddr_in6`), but you can still refer to the member via `enftun_tcp.local_addr`.